src/components/ClientsListContainer.tsx

```diff
@@ -35,16 +51,29 @@ export const ClientsListContainer: FC<Props> = ({ setClientID }) => {
     fetchClients();
   }, []);
 
-  const handelSearch = (search: string): void => {
+  const searchClients = useDebouncedCallback((search: string): void => {
     dispatch(setSearch(search));
+  }, 800);
+
+  const handelSearch = (search: string): void => {
+    setClientSearch(search);
+    searchClients(search);
   };
 
+  if (client && isHideClientsList) return null;
   if (isClientsLoading) return <CircularProgress className={classes.load} />;
```

Comment:
I think you need to set `CircularProgress` before checking `client` and `isHideClientsList`

Like this:
`if (isClientsLoading) return <CircularProgress className={classes.load} />;`
`if (client && isHideClientsList) return null;`

Reply:
Done